app.go

```diff
@@ -22,10 +23,15 @@ func main() {
 		log.Fatal(err)
 	}
 
+	var err error
 	// Start the server.
 	if os.Getenv("HTTP_PLATFORM_PORT") != "" {
-		http.ListenAndServe(":"+os.Getenv("HTTP_PLATFORM_PORT"), nil)
+		err = http.ListenAndServe(":"+os.Getenv("HTTP_PLATFORM_PORT"), nil)
 	} else {
-		http.ListenAndServe(":8080", nil)
+		err = http.ListenAndServe(":8080", nil)
+	}
+
+	if err != nil {
+		log.Fatal(err)
 	}
```

Comment:
As we are not doing any error handling other than `log.Fatal`, we could change every `http.ListenAndServe()` call to `log.Fatal(ListenAndServe())`.

Reply:
Agree

Reply:
No, I disagree, it makes it less clear what's going on.
What if http.ListenAndServe returns nil (i.e. no-error)?

Reply:
If it returns nil, we just stop the app. 

First of all, we have to understand how `http.ListenAndServe` works and what is the "purpose" of it. When it is called, it blocks and waits forever for clients (so, in our case, it is the "main" thread). If `http.ListenAndServe` returns an `error` or `nil` it means that it has been stopped. From here we have 2 possibilities:
1. If it returned `nil`, we just stop the app (since `log.Fatal()` will not print anything and will do `os.Exit(1)`);
2. If it returned `error`, we log it with `log.Fatal` and exit our app.

In conclusion, if, for whatever the reason, `http.ListenAndServe` stopped, we want to exit as this was our "main" thread of our app.

Reply:
I disagree, log.Fatal(http.ListenAndServe) is unclear, since it's unclear that http.ListenAndServe blocks. 
Let's keep it simple.
What if we ever wanted to have something happen after server closes?